### PR TITLE
Add support for GPUs out-of-the-box

### PR DESCRIPTION
**Pull request by amandineslx**
[sc-96860]

### CODE REVIEW

**Comment by jules-casoli** [comment]
> This will not work in air-gapped deployments. Maybe loop back with FE team to check if air-gapped GPU deployments are to be expected soon and should be scoped for this PR.
We are mostly doing the same as what is done for aws/gcp, so I guess it don't work there also ?
We could cache the daemonSet yaml locally, but still, we'll need the nvidia driver images to be available on the client's repo. Should we trust them on doing it ? I guess they would have similar problem in air-gapped envs. with the code-envs, do you know how they handle that ?

**Comment by jules-casoli** [comment]
Milestone is probably rather 11.2.0

File: Changelog.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Next version
+
+- Support GPU out-of-the-box
```
Comment (NawreCode) [inline Changelog.md:5]: Could be just `Support for GPU` 😉

**Review by NawreCode — APPROVED** [review]
LGTM. Now works after a few clicks without too much hassle.

What about air-gapped environments? I guess it's not working for the other plugins either. It's out-of-scope for this PR anyway.

**Review by jules-casoli — APPROVED** [review]
obviously, LGTM

**Comment by duckie** [comment]
> > This will not work in air-gapped deployments. Maybe loop back with FE team to check if air-gapped GPU deployments are to be expected soon and should be scoped for this PR.
> 
> 
> 
> We are mostly doing the same as what is done for aws/gcp, so I guess it don't work there also ?
> 
> We could cache the daemonSet yaml locally, but still, we'll need the nvidia driver images to be available on the client's repo. Should we trust them on doing it ? I guess they would have similar problem in air-gapped envs. with the code-envs, do you know how they handle that ?
> 
> 

Indeed that cannot work here either, but then we have much more customers on Azure, hence the advice about wether that should be scoped or not. I guess not and let's see later with customers.

Maybe the intersection of gpu users and air gapped users is very small.

**Pull request merged**